src/components/organisms/NavigatorPane/ResourceNavigator.tsx

```diff
@@ -0,0 +1,102 @@
+import {useRef} from 'react';
+
+import {Skeleton} from 'antd';
+
+import styled from 'styled-components';
+
+import {useAppSelector} from '@redux/hooks';
+import {resourceNavigatorSelector} from '@redux/selectors/resourceSelectors';
+
+import {elementScroll, useVirtualizer} from '@tanstack/react-virtual';
+
+import EmptyResourceNavigator from './EmptyResourceNavigator';
+import KindRenderer from './KindRenderer';
+import ResourceRenderer from './ResourceRenderer';
+import {useScroll} from './useScroll';
+
+function ResourceNavigator() {
+  const list = useAppSelector(resourceNavigatorSelector);
+  const ref = useRef<HTMLUListElement>(null);
+  const isLoading = useAppSelector(
+    state =>
+      state.ui.isFolderLoading || state.main.previewOptions.isLoading || state.main.clusterConnectionOptions.isLoading
+  );
+
+  const rowVirtualizer = useVirtualizer({
+    count: list.length,
+    estimateSize: () => 28,
+    getScrollElement: () => ref.current,
+    scrollToFn: elementScroll,
+  });
+
+  useScroll({
+    list,
+    scrollTo: index =>
+      rowVirtualizer.scrollToIndex(index, {
+        align: 'center',
+        behavior: 'smooth',
+      }),
+  });
+
+  if (list.length === 0) {
+    return <EmptyResourceNavigator />;
+  }
+
+  if (isLoading) {
+    return (
+      <div style={{padding: '10px'}}>
+        <Skeleton />
+      </div>
+    );
+  }
+
+  return (
+    <ListContainer ref={ref}>
+      <div
```

Comment:
Can we create a styled component for this?

Reply:
We can't, it breaks the virtualization if we do that